x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java

```diff
@@ -330,13 +332,17 @@ public boolean functionExists(String name) {
     static <T extends Function> FunctionDefinition def(Class<T> function,
             BinaryFunctionBuilder<T> ctorRef, String... aliases) {
         FunctionBuilder builder = (location, children, distinct, tz) -> {
-            if (children.size() != 2) {
+            boolean isBinaryOptionalParamFunction = function.isAssignableFrom(Round.class) || function.isAssignableFrom(Truncate.class);
```

Comment:
The handling of 2 vs 1 parameter should not be handled here or at least this way - it should either be generic (so it doesn't care about the name) or is handled inside the Analyzer#ResolveFunctions. To not stall this any further, please raise another issue to have this as a follow-up.

Reply:
Acked. Issue created: https://github.com/elastic/elasticsearch/issues/33885